dask_sql/server/presto_jdbc.py

```diff
@@ -0,0 +1,175 @@
+
+    # if we supply a catalogue dbeaver (for instance) tries to use it
+    # but dask_sql presto interface doesn't support catalog
+    # so better for now not to supply a catalog
+    # if catalog and len(catalog.strip()) > 0:
+    #     catalogs = pd.DataFrame().append(create_catalog_row(catalog), ignore_index=True)
+    #     c.create_table("catalogs", catalogs, schema_name=system_schema)
+
+    schemas = pd.DataFrame().append(create_schema_row(), ignore_index=True)
+    c.create_table("schemas", schemas, schema_name=system_schema)
+    schema_rows = []
+
+    tables = pd.DataFrame().append(create_table_row(), ignore_index=True)
+    c.create_table("tables", tables, schema_name=system_schema)
+    table_rows = []
+
+    columns = pd.DataFrame().append(create_column_row(), ignore_index=True)
+    c.create_table("columns", columns, schema_name=system_schema)
+    column_rows = []
+
+    for schema_name, schema in c.schema.items():
+        schema_rows.append(create_schema_row(catalog, schema_name))
+        for table_name, dc in schema.tables.items():
+            df = dc.df
+            logger.info(f"schema ${schema_name}, table {table_name}, {df}")
+            table_rows.append(create_table_row(catalog, schema_name, table_name))
+            pos: int = 0
+            for column in df.columns:
+                pos = pos + 1
+                logger.debug(f"column {column}")
+                dtype = "VARCHAR"
+                if df[column].dtype == "int64" or df[column].dtype == "int":
+                    dtype = "INTEGER"
+                elif df[column].dtype == "float64" or df[column].dtype == "float":
+                    dtype = "FLOAT"
+                elif (
+                    df[column].dtype == "datetime"
+                    or df[column].dtype == "datetime64[ns]"
+                ):
+                    dtype = "TIMESTAMP"
+                column_rows.append(
+                    create_column_row(
+                        catalog,
+                        schema_name,
+                        table_name,
+                        dtype,
+                        df[column].name,
+                        str(pos),
+                    )
+                )
+
+                # print(f'col {df[column].name} {df[column].dtype}')
+                # print(f'crow dtypes {crow.dtypes}')
+                # print(f'columns dtypes {columns.dtypes}')
+
+    schemas = pd.DataFrame(schema_rows)
+    c.create_table("schemas", schemas, schema_name=system_schema)
+    tables = pd.DataFrame(table_rows)
+    c.create_table("tables", tables, schema_name=system_schema)
+    columns = pd.DataFrame(column_rows)
+    c.create_table("columns", columns, schema_name=system_schema)
+
+    logger.info("jdbc meta data ready")
+
+
+def create_catalog_row(catalog: str = ""):
+    return {"TABLE_CAT": catalog}
+
+
+def create_schema_row(catalog: str = "", schema: str = ""):
+    return {"TABLE_CATALOG": catalog, "TABLE_SCHEM": schema}
```

Comment:
This is what the presto driver expects. I thought it odd too

Reply:
Ah okay thanks for the clarification - if Trino is the more active fork of Presto, perhaps it would be worthwhile to explore if it is possible to use here instead of PrestoDB, though there is no push to do that before merging this in